<p>There is no good excuse for an empty class. If it's being used simply as a common extension point, it should be replaced with an {{interface}}. If it was stubbed in as a placeholder for future development it should be fleshed-out. In any other case, it should be eliminated.</p>

<h2>Noncompliant Code Example</h2>
<pre>
public class Nothing {  // Noncompliant
}
</pre>

<h2>Compliant Solution</h2>
<pre>
public interface Nothing {
}
</pre>